MdePkg/Include/IndustryStandard/Acpi65.h

```diff
@@ -2944,6 +2944,54 @@ typedef struct {
 #define EFI_ACPI_6_5_PHAT_FIRMWARE_HEALTH_DATA_RECORD_UNKNOWN          0x02
 #define EFI_ACPI_6_5_PHAT_FIRMWARE_HEALTH_DATA_RECORD_ADVISORY         0x03
 
+///
+/// Reset Reason Health Record Vendor Data Entry
+///
+typedef struct {
+  EFI_GUID  VendorDataID;
+  UINT16    Length;
+  UINT16    Revision;
+  // UINTN   Data[];
+} EFI_ACPI_6_5_PHAT_RESET_REASON_HEALTH_RECORD_VENDOR_DATA_ENTRY;
+
+///
+/// Reset Reason Health Record Structure
+///
+typedef struct {
+  UINT8     SupportedSources;
+  UINT8     Source;
+  UINT8     SubSource;
+  UINT8     Reason;
+  UINT16    VendorCount;
+  // EFI_ACPI_6_5_PHAT_RESET_REASON_HEALTH_RECORD_VENDOR_DATA_ENTRY   VendorSpecificResetReasonEntry[];

```

Comment:
edk2 supports C structure flexible array members.  Is there a reason that is not used here?

Reply:
just following the implementation used for other structures in this file. Refer EFI_ACPI_6_5_PHAT_FIRMWARE_HEALTH_DATA_RECORD_STRUCTURE